cli/src/utils/getdeps.ts

```diff
@@ -8,8 +15,18 @@ export async function getDependencies(projectDir: string) {
   const { dependencies: lockDeps } = JSON.parse(await readFile(packageLockPath, 'utf8'));
   const { dependencies: pkgDeps } = JSON.parse(await readFile(packageJsonPath, 'utf8'));
 
-  const dependencies = new Set<string>();
   const toProcess = Object.keys(pkgDeps);
+
+  // If package.json contains packages that are not in
+  // package-lock.json then they are out of sync.  Complain.
+  const lockHasProperty = Object.prototype.hasOwnProperty.bind(lockDeps);
```

Comment:
I don't really like `Object.prototype.hasOwnProperty` (since it's so long to read and write). IMO something like `lodash.has` is at least much shorter to read (even if the implementation is the same).

I think we should avoid trying to prototype-proof _all_ the code, there will always be a piece of code that behaves in strange ways when encountering unusual objects. This code _would_ correctly protect against the case if we installed [prototypes](https://www.npmjs.com/package/prototypes) and then running on a `package.json` that has a dependency on [concat](https://www.npmjs.com/package/concat), but since we have some control over our dependencies we can try not to ones that modify the object prototype.